imsv-docs-astro/src/content/docs/guides/funding-protocols/flexi-algorand-funding-protocol.mdoc

```diff
@@ -0,0 +1,226 @@
+**Note**: This functionality is currently available for Developer Preview only
+and not for general use.
+{% /warning %}
+
+The Immersve Flexi Algorand {% link page="guides/funding-protocols"
+title="funding protocol" /%} allows client applications to fund Immersve cards
+via simple native Algorand {% link
+href="https://developer.algorand.org/docs/get-details/asa" title="Asset" /%}
+transfers.
+
+Depositors can withdraw funds at any time with signed withdrawal approvals.
+Deposits are scoped to a single Immersve partner application. Depositor funds
+can only be settled to Immersve's approved settlement address.
+
+## Smart Contract
+
+The Flexi Algorand protocol has a "Master" contract and many partner "Partner
+Channels" contracts. Partner Channel contracts are used to keep track of the
+funds associated to the cardholders under a specific partner. The cardholder
+funds are stored in a "Card Fund" contract. Both "Partner Channels" and "Card
+Funds" are managed by the "Master" Contract
+
+
+### Source
+{% link href="https://github.com/immersve/funding-contract-algorand" /%}.
+
+### Deployed Contract Addresses
+
+The Flexi Algorand protocol is designed to work on the Algorand networks. The deployed
+contract addresses are listed below.
+
+{%
+  deployedFundingProtocolsTable
+  protocol="flexi-algorand"
+  columns=[
+    "network",
+    "address",
+  ]
+/%}
+
+### Smart Contract Overview
+
+The Master contract is the main entry point and is a factory for the Partner
+Channel and Card Funds contracts. Card Fund contracts are the target for
+depositor native asset transfers that are scoped to an integrated partner
+application. The Master contract is also used by Immersve for performing
+settlements and refunds.
+
+The Partner Channel contract is used to identify the partner to which a Card
+Fund contract belongs. The Partner Channel is provisioned by the Immersve card
+issuing partner.
+
+The Card Fund contract holds assets for funding a cardholders card. Deposits are
+made via native token transfers. The card fund is provisioned by the cardholder.
+A card fund contract is uniquely identified by the address of the owner (the
+cardholder's primary algorand address) and the address of the related partner
+contract.
+
+![flexi-algorand-overview](./assets/algorand-component-diagram.png "Algorand
+smart contract component diagram.")
+
+Upon receiving a card payment request, a Card Fund contract will be debited by
+Immersve. The debited funds move to the Master Contract where they wait to be
+cleared for settlement.
+
+![flexi-algorand-overview](./assets/algorand-funds-flow-diagram.png "Algorand
+funds flow diagram.")
+
+
+## Protocol Mechanics
+
+### Deposits
+
+Depositors can send native asset transfers directly to the Card Funds address.
+Each depositor address needs to have a [Funding
+Source](https://docs.immersve.com/api-reference/create-a-funding-source-for-an-account)
+created that is linked with the Card Fund. After an asset transfer
```

Comment:
The Funding Source needs a one line introduction or at least some language to indicate it is an off-chain resource without the reader going to the endpoint. The endpoint doesn't introduce it that well anyway.